Frameworks/AudioToolbox/SystemSound.mm

```diff
@@ -16,62 +17,273 @@
 
 #import <AudioToolbox/SystemSound.h>
 #import <StubReturn.h>
+#import <Starboard.h>
+#import <UWP/WindowsMediaPlayback.h>
+#import <mutex>
+
+
+// Calls into C++ to access Windows.Phone.Devices.Notification namespace to vibrate devices with vibration capability.
+void vibrate();
+
+static const wchar_t* TAG = L"AudioToolbox";
+
+// We generate SystemSoundIDs starting with 5001 as some system supplied sounds have ids below 5000.
+static const SystemSoundID c_userSSID = 5000;
+
+// We use the NSMutableDictionary to map SystemSoundID to URL via a SystemSound object.
+NSMutableDictionary* _ssObjects = [NSMutableDictionary dictionary];
```

Comment:
nit: Prefer that you don't use abbreviations where possible. Though the meaning may be obvious to you, it is unlikely that you will be the only author / contributor to this code. Abbreviations add to a sense of "tribal knowledge" that discourages casual readers from understanding and being able to contribute.